src/emailmessage.cpp

```diff
@@ -17,6 +17,25 @@
 #include "qmailnamespace.h"
 #include <qmaildisconnected.h>
 #include <QTextDocument>
+#include <QTemporaryFile>
+#include <QStandardPaths>
+#include <QDir>
+
+// Part finder
+struct PartFinder {
+    PartFinder(QMailMessageContentType type, const QMailMessagePart *&part) : contentType(type), partFound(part) {}
```

Comment:
I think it would be cleaner to have 'partFound' as a read-only property of the finder, rather than to pass it in to the ctor.